data/web/inc/spf.inc.php

```diff
@@ -1,6 +1,8 @@
 <?php
 error_reporting(0);
-function get_spf_allowed_hosts($check_domain, $expand_ipv6 = false, $original_domain = "") {
+$checked_domains = array();
+function get_spf_allowed_hosts($check_domain, $expand_ipv6 = false) {
+	global $checked_domains;
```

Comment:
Please don't use global variables. This fill break the loop detection when multiple SPF checks are performed on the same page.

Reply:
Sorry, for some reason I didn't think about just passing the array back into the function.
I've changed it to pass in a new optional argument, the array containing all previously checked include or redirect SPF domain records.